### PR TITLE
18zoo - Implements fixes asked inside https://github.com/tobymao/18xx…

### LINKED ISSUE #10462: [18ZOO] selling not recorded - work in progress doesn't work properly
**What's your Game ID?**
`https://18xx.games/game/153535`

**How to Reproduce the Bug?**
*selling not recorded*
https://18xx.games/game/153535?action=96
- Previous step Serpentium sold PE
- Now PE is not recorded as sold, so Serpentium can buy again 
image

The expected behaviour is that item sold should be blocked
image

*work in progress doesn't work properly*
https://18xx.games/game/153535?action=198
image
- in the action 199 Serpentium used "Work in progress" and instead of going to second slot, it overrides the first city slot
image

The expected behaviour is that item goes to second slot
image

**Which Browser / Platform?**

### PR DESCRIPTION
**Pull request by carlorusso1984**
Fixes #10462

## Before clicking "Create"

- [x] Branch is derived from the latest `master`
- [x] Add the `pins` or `archive_alpha_games` label if this change will break existing games
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

### CODE REVIEW

File: lib/engine/player.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def corporation
     end
 
     def ==(other)
-      other&.player? && (@name == other&.name)
+      other&.player? && !other.player.nil? && (@name == other.name)
```
Comment (tobymao) [inline lib/engine/player.rb:44]: why was this needed?

Comment (tobymao) [reply lib/engine/player.rb:44]: you shouldn't call nil? other.playre is enough

Comment (carlorusso1984) [reply lib/engine/player.rb:44]: As this discussion: https://18xxgames.slack.com/archives/C012K0CNY5C/p1708883331307309?thread_ts=1708871941.895459&cid=C012K0CNY5C
the issue is that in Javascript this code `other&.player?`  is translated into a check that player is not null, but in that case the Player is NilClass, so it's returning true also if the player is null
With this fix, the problem is solved also on JS

Comment (tobymao) [reply lib/engine/player.rb:44]: can you do 

`!!other&.player? && name == other.name`

Comment (carlorusso1984) [reply lib/engine/player.rb]: done in d293b0f

**Review by ollybh — APPROVED** [review]

**Pull request merged**